pkg/apis/operator/v1alpha1/tektonchain_validation.go

```diff
@@ -48,7 +49,18 @@ func (tc *TektonChain) Validate(ctx context.Context) (errs *apis.FieldError) {
 		errs = errs.Also(apis.ErrMissingField("spec.targetNamespace"))
 	}
 
-	return errs.Also(tc.Spec.ValidateChainConfig("spec"))
+	return errs.Also(tc.Spec.ValidateControllerEnv(), tc.Spec.ValidateChainConfig("spec"))
+}
+
+func (tcs *TektonChainSpec) ValidateControllerEnv() (errs *apis.FieldError) {
+	if tcs.ControllerEnvs != nil {
+		for _, v := range tcs.ControllerEnvs {
+			if !allowedControllerEnvs.Has(v.Name) {
+				errs = errs.Also(apis.ErrInvalidKeyName(v.Name, "supported key is `MONGO_SERVER_URL`"))
```

Comment:
IMHO, 
may be, no need to touch in the future, if we include supported fields dynamically.
```
errs = errs.Also(apis.ErrInvalidKeyName(v.Name, fmt.Sprintf("supported keys are %s",strings.Join(allowedControllerEnvs.List(), ","))))
```